opm/simulators/flow/Main.hpp

```diff
@@ -344,7 +340,18 @@ class Main
         }
 
         if (enableDamarisOutput_) {
-            this->setupDamaris(outputDir); // Damaris server ranks will block here until damaris_stop() is called by client ranks
+            // Deal with empty (defaulted) output dir, should be deck dir
+            auto damarisOutputDir = outputDir;
+            if (outputDir.empty()) {
+                const auto slash_pos = deckFilename.find_last_of("/\\");
```

Comment:
Will this work if a file name contains spaces? Those would probably be escaped, e.g. part1\ part2\deck.data

Reply:
I am not sure. But this is the same logic as in Deck::setDataFile() so if wrong we need to address both places.

Reply:
That looks a lot more robust. I will amend here.